### PR TITLE
[MUMPS_seq] new formula v5.2.1

### PR DESCRIPTION
**Pull request by dpo**
🤞

### CODE REVIEW

File: M/MUMPS_seq/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+platforms = expand_gfortran_versions(filter!(p -> !isa(p, Windows) && arch(p) != :powerpc64le, supported_platforms()))
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libsmumps", :libsmumps),
+    LibraryProduct("libdmumps", :libdmumps),
+    LibraryProduct("libcmumps", :libcmumps),
+    LibraryProduct("libzmumps", :libzmumps),
+]
```
Comment (ViralBShah) [inline M/MUMPS_seq/build_tarballs.jl:93]: I think these names are the same as the parallel package.

**Comment by ViralBShah** [comment]
Looking at coinbrew for how the whole ecosystem is built (on recommendation from the authors) it looks like they use [Mumps 4.10](https://github.com/coin-or-tools/ThirdParty-Mumps/blob/8f81226225a53db1e86d600f4e29f93f2b6f5bff/get.Mumps#L5).

**Comment by dpo** [comment]
I've used IPOPT with this newer version of MUMPS. Does COIN use MUMPS for anything else than IPOPT?

**Comment by ViralBShah** [comment]
Clp has the ability to use MUMPS. I suspect if the API is compatible it might just work. Should be easy to try before bundling up another version.

File: M/MUMPS_seq/build_tarballs.jl (hunk 1 above)
Comment (dpo) [reply M/MUMPS_seq/build_tarballs.jl:93]: Isn't it a problem that they define the same symbols though?

**Comment by dpo** [comment]
🤘

File: M/MUMPS_seq/build_tarballs.jl (hunk 1 above)
Comment (ViralBShah) [reply M/MUMPS_seq/build_tarballs.jl:93]: Apparently not. @giordano @staticfloat Can you guys correct me if I am wrong?

Comment (giordano) [reply M/MUMPS_seq/build_tarballs.jl:93]: As it is now, I think `MUMPS_jll` and `MUMPS_seq_jll` won't be able to coexist in the same environment.  On the other hand mangling the name of the library file adds more complexity for all the libraries that will need to link against this one in particular, see the mess we always need to do for OpenBLAS.

**Comment by amontoison** [comment]
Maybe sequential MUMPS works on Windows? The problem was MPI with the parralel version.

**Comment by dpo** [comment]
Let's try.

**Comment by dpo** [comment]
The builders are building MUMPS and not MUMPS_seq. Any idea what's going on here?

**Comment by giordano** [comment]
> The builders are building MUMPS and not MUMPS_seq. Any idea what's going on here?

In the last commit you edited only `M/MUPS/build_tarballs.jl`, so only this one was actually built, for `M/MUPS_seq` the cache is still hot.  You can see in the [`generator`](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=2501&view=logs&jobId=e2fb71e5-f3df-5cf3-8f73-e5a3a93ebd57&j=96c13d9d-dfd2-51e5-2f76-4a0b3ce09abc&t=09532434-8b0a-5f0e-118d-a3181cde5ec8) job what's happening:
```
Considering M/MUMPS
 --> Accepted!
Considering M/MUMPS_seq
 --> Accepted!
[...]
Determining builds to queue...
 ---> MUMPS: 585c625152a5fead252a068015abcd09a253ab29/bbc1bd0ab96bc83a7307e14c34edbd84c0836d3996a70528df57a927e1936feb (0cd6beb091eed9d6a73dc15d41456369c1cad2ba + f702ae53dc68d266ed1b98e2aae9030944f445f076453b7c3d49e6aad6a15fc6)
    i686-linux-gnu-libgfortran3: building
    i686-linux-gnu-libgfortran4: building
    x86_64-apple-darwin14-libgfortran4: building
    x86_64-apple-darwin14-libgfortran5: building
    x86_64-unknown-freebsd11.1-libgfortran3: building
    x86_64-unknown-freebsd11.1-libgfortran4: building
    x86_64-unknown-freebsd11.1-libgfortran5: building
    i686-w64-mingw32-libgfortran3: building
    i686-w64-mingw32-libgfortran4: building
    i686-w64-mingw32-libgfortran5: building
    x86_64-w64-mingw32-libgfortran3: building
    x86_64-w64-mingw32-libgfortran4: building
    x86_64-w64-mingw32-libgfortran5: building
 ---> MUMPS_seq: 585c625152a5fead252a068015abcd09a253ab29/fe3b03e9bd6c3672e2112097b9f94ccfcd8d0a1fd79b9d8bb4b65ff5ad43f95c (f5cb068f534366d95e7f149fa7098328d03626fa + 01f5b6ce013cbadd61cda531b2bea4de5ad2896e8ef2e7d7cc4ef8168756f828)
    i686-linux-gnu-libgfortran3: skipping, existant
    i686-linux-gnu-libgfortran4: skipping, existant
    i686-linux-gnu-libgfortran5: skipping, existant
    x86_64-linux-gnu-libgfortran3: skipping, existant
    x86_64-linux-gnu-libgfortran4: skipping, existant
    x86_64-linux-gnu-libgfortran5: skipping, existant
    aarch64-linux-gnu-libgfortran3: skipping, existant
    aarch64-linux-gnu-libgfortran4: skipping, existant
    aarch64-linux-gnu-libgfortran5: skipping, existant
    armv7l-linux-gnueabihf-libgfortran3: skipping, existant
    armv7l-linux-gnueabihf-libgfortran4: skipping, existant
    armv7l-linux-gnueabihf-libgfortran5: skipping, existant
    i686-linux-musl-libgfortran3: skipping, existant
    i686-linux-musl-libgfortran4: skipping, existant
    i686-linux-musl-libgfortran5: skipping, existant
    x86_64-linux-musl-libgfortran3: skipping, existant
    x86_64-linux-musl-libgfortran4: skipping, existant
    x86_64-linux-musl-libgfortran5: skipping, existant
    aarch64-linux-musl-libgfortran3: skipping, existant
    aarch64-linux-musl-libgfortran4: skipping, existant
    aarch64-linux-musl-libgfortran5: skipping, existant
    armv7l-linux-musleabihf-libgfortran3: skipping, existant
    armv7l-linux-musleabihf-libgfortran4: skipping, existant
    armv7l-linux-musleabihf-libgfortran5: skipping, existant
    x86_64-apple-darwin14-libgfortran3: skipping, existant
    x86_64-apple-darwin14-libgfortran4: skipping, existant
    x86_64-apple-darwin14-libgfortran5: skipping, existant
    x86_64-unknown-freebsd11.1-libgfortran3: skipping, existant
    x86_64-unknown-freebsd11.1-libgfortran4: skipping, existant
    x86_64-unknown-freebsd11.1-libgfortran5: skipping, existant
```

**Comment by dpo** [comment]
Argh. Got mixed up. Sorry.

**Comment by dpo** [comment]
There we go.

**Comment by dpo** [comment]
Should be fixed.

**Comment by dpo** [comment]
MUMPS 5.3.0 came out today, but we'd better test with this one because the API hasn't changed in a while, and I'm not sure yet what changed in 5.3.

**Pull request merged**